hooks/open-telemetry/README.md

```diff
@@ -1,38 +1,92 @@
 # OpenTelemetry Hook
 
-### Requirements
+## Requirements
 
 - open-feature/go-sdk >= v1.3.0
 
 ## Usage
 
-For this hook to function correctly a global `TracerProvider` must be set, an example of how to do this can be found below.
+## Metric hook
 
-The `open telemetry hook` taps into the after and error methods of the hook lifecycle to write `events` and `attributes` to an existing `span`.
+This hook performs metric collection by tapping into various hook stages. Given below are the metrics are extracted 
+by this hook,
+
+- `feature_flag.evaluation_requests_total`
+- `feature_flag.evaluation_success_total`
+- `feature_flag.evaluation_error_total`
+- `feature_flag.evaluation_active_count`
+
+### Options
+
+#### WithFlagMetadataDimensions 
+
+This constructor option allows to configure dimension descriptions to be extracted from `openfeature.FlagMetadata`. 
+If present, these dimension will be added to the `feature_flag.evaluation_success_total` metric.
+
+Note that, this configuration must be carefully coordinated with `Provider` as missing dimensions results in hook 
```

Comment:
Does that mean the hook will through if `scope` is missing from the hook metadata?

Reply:
By default (no configurations) nothing will happen. See this test case as an example [1]. However, if someone configures the hook with  `WithFlagMetadataDimensions`, similar example below,

```go
NewMetricsHook(reader,
    WithFlagMetadataDimensions(
        DimensionDescription{
            Key:  "scope",
            Type: String,
        }))
```

then the logic will try to extract the flag metadata entry, which returns an error. And this error comes from go-sdk itself , see GetString for example [2]

[1] - https://github.com/open-feature/go-sdk-contrib/pull/217/files#diff-4947cec855aa6f6fb207103527911ad05567255efcc5f569bbcc0d3ffe80a17fR57-R109
[2] - https://github.com/open-feature/go-sdk/blob/main/pkg/openfeature/client.go#L176

Reply:
Updated the implementation to ignore missing metadata and avoid error returns. I think this is the least complex implementation.